core/src/AWS/Lambda/Handler.hs

```diff
@@ -1,31 +1,51 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecordWildCards   #-}
 
-module AWS.Lambda.Handler where
+module AWS.Lambda.Handler
+  ( handler
+  , handle
+  ) where
 
 import           AWS.Lambda.Encoding
 import           AWS.Lambda.RuntimeClient
-import           AWS.Lambda.Types
 import           Control.Monad
-import           Control.Monad.Except
-import           Control.Monad.IO.Class
 import           Control.Monad.Logger
 import qualified Data.Aeson as A
-import qualified Data.ByteString as BS
-import qualified Data.ByteString.Lazy as LBS
-import           Data.Text
-import qualified Data.Text.Encoding as TE
+import qualified Data.Text as T
+import           UnliftIO hiding (handle)
 
-handler :: (LambdaEncode err, MonadError err m, MonadIO m, MonadLogger m, LambdaDecode e, LambdaEncode r) => (e -> m r) -> m ()
+
+-- | Loop forever handling events with a default runtime client and a given callback
+handler
+  :: MonadUnliftIO m
+  => MonadLogger m
+  => LambdaDecode e
+  => LambdaEncode r
+  => (e -> m r)
+  -> m ()
 handler f = forever $ do
   client <- runtimeClient
   forever $ handle client f
 
-handle :: (LambdaEncode err, MonadError err m, MonadIO m) => RuntimeClient e r m -> (e -> m r) -> m ()
-handle c@RuntimeClient{..} f = do
-  event@Event{..} <- getNextEvent
-  let
-    result = case eventBody of
-      Right val -> f val >>= postResponse eventID
-      Left e    -> postError eventID (Error "Parse failure" $ (A.String . pack) e)
-  catchError result (postError eventID . Error "Error" . encodeOutput)
+
+-- | Handle single event with a given runtime client and a callback
+handle
```

Comment:
I'm not even sure we should expose `handle` to be honest. `handler` should be the only API used by apps. This is a runtime, not a simple function, and the runtime does not know how long it will be alive (AWS can send 1 or more events to a runtime, depending on how long the function is being kept "hot" in the backend), and this will crash if it only handled 1 event. In the examples we only use `handler` for this reason and it was a mistake on my part exposing `handle` as well, as this leads to an incorrect implementation.

Reply:
Yeah, I was wondering why we expose both, thought it could be so that you'd have a chance to customize runtime somehow, e.g. tweak http client settings or some such.
If it's not needed then yeah - one function is better and I'd actually like to rename it to `handle` then.

Reply:
I used `handler` because it seems pretty standard (other runtimes like https://github.com/Nike-Inc/hal and https://theam.github.io/aws-lambda-haskell-runtime/ use the same term). I don't think it makes much sense to customize the current client (we don't expose anything special) but if so, i would rename `handle` to something like `handlerWIthClient` (find a better name :P )